core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/PooledDatasourceHelper.java

```diff
@@ -332,7 +332,12 @@ public static DataSource convert( IDatabaseConnection databaseConnection ) throw
       String driverClassName =
           databaseConnection.getAttributes().get( GenericDatabaseDialect.ATTRIBUTE_CUSTOM_DRIVER_CLASS );
       if ( !StringUtils.isEmpty( driverClassName ) ) {
-        basicDatasource.setDriverClassName( driverClassName );
+        try {
+          basicDatasource.setDriverClassName( driverClassName );
+        } catch ( Throwable th ) {
+          throw new DBDatasourceServiceException( Messages.getInstance().getErrorString(
```

Comment:
We're losing the root cause exception here. Extend DBDatasourceServiceException to add a constructor that takes a Throwable cause